### ISSUE TITLE
CredentialsProviderError when integrating AXIOM 

### ISSUE THREAD

**Issue by vucuong** [serverless/serverless#12874]
### Issue description

I integrated AXIOM in following the instructions at https://www.serverless.com/framework/docs/guides/observability/axiom. 

serverles.yml 
```
stages:
    dev:
        observability:
            provider: axiom
            dataset: dev-cloudwatch

    prod:
        observability:
            provider: axiom
            dataset: prod-cloudwatch
         
     default:
        observability: false
            
provider:
    name: aws
    runtime: nodejs20.x
    stage: dev
    profile: deploy_${opt:stage, self:provider.stage}            
```

The error occurred after the console appeared "⠋ Integrating with Axiom"

```
✖ Stack user-service-dev failed to deploy

✖ CredentialsProviderError: Could not load credentials from any providers
    at file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:182:1742
    at file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:77:2711
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async coalesceProvider (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:77:3083)
    at async file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:77:3372
    at async file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:76:67235
    at async file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:76:14549
    at async createForwarder (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:949:4575)
    at async initialize (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:951:88265)
    at async Axiom2.subscribeLogGroups (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:951:92431)
    at async PluginManager.runHooks (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:953:9311)
    at async PluginManager.invoke (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:953:10126)
    at async PluginManager.run (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:953:10813)
    at async Serverless.run (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:960:10612)
    at async runFramework (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:983:2696)
    at async route (file:///Users/my-pc/.serverless/releases/4.4.6/package/dist/sf-core.js:1011:13954)
```

After checking, I found that it only seems to receive the [default] credentials from ~/.aws/credentials. However, I'm using deploy_dev and deploy_prod, and the above error occurs. When I remove the AXIOM integration, it works.

**Comment by czubocha** [comment]
Hey @vucuong, thanks for reporting this. We'll include the fix in the next release.

**Comment by czubocha** [comment]
Hey @vucuong, the fix has been released in version [4.4.7](https://github.com/serverless/serverless/releases/tag/v4.4.7). Feel free to let us know if you encounter any further issues.
